spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SessionsController, type: :controller do
+  describe 'GET#new' do
+    it 'renders a successful response' do
+      get :new
+
+      expect(response).to be_successful
+    end
+
+    it 'renders a correct template' do
+      get :new
+
+      expect(response).to render_template(:new)
+    end
+
+    it 'redirect to home when session exist' do
+      user = Fabricate(:user, username: 'nimblehq', password: 'password', password_confirmation: 'password')
+      session[:user_id] = user[:id]
+
+      get :new
+
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe 'POST#create' do
+    context 'with valid parameters' do
+      it 'username and password are valid' do
+        user = Fabricate(:user, username: 'nimblehq', password: 'password', password_confirmation: 'password')
+
+        post :create, params: { username: user[:username], password: 'password' }
+
+        expect(session[:user_id]).to eql(user[:id])
+        expect(response).to redirect_to(root_path)
+      end
+    end
+
+    context 'with invalid parameters' do
+      before(:each) do
+        Fabricate(:user, username: 'nimblehq', password: 'password', password_confirmation: 'password')
+      end
+
+      # Every test cases expect the same thing
+      after(:each) do
```

Comment:
we don't use after hook @gutakk the reason is we keep the test is isolate and easy to read, maintain later.

Reply:
Fixed in 39bbfb7995826c54e37ae6c741a590e2dad66850